### PR TITLE
Preparations for use of ddrgen

### PR DESCRIPTION
**Pull request by keithc-ca**
Preparations for use of ddrgen in support of openj9#378

* correct references to getmacros script
* add missing ddr_namespace annotations
* use J9HASH_TABLE_AVL_TREE_TAG_BIT instead of hard-coded constant
* ignore objs directory

### CODE REVIEW

**Comment by fjeremic** [comment]
Cross linking the OpenJ9 issue https://github.com/eclipse/openj9/issues/378.

File: ddr_artifacts.mk [hunk 1]
```diff
@@ -30,10 +30,10 @@ top_srcdir = .
 include $(top_srcdir)/omrmakefiles/configure.mk
 
 # Ensure the source tree path is canonically-formed and ends with /
-abs_srcroot = $(abspath $(TOP_SRCDIR))/
+abs_srcroot := $(subst //,/,$(abspath $(TOP_SRCDIR))/)
 
 all: check-vars
-	./tools/ddrgen/src/macros/getMacros.sh $(abs_srcroot)
+	bash $(top_srcdir)/ddr/tools/getmacros $(abs_srcroot)
```
Comment (charliegracie) [inline ddr_artifacts.mk]: should we be relying on `bash`?  Can we not just use `sh`?  I know there are sometimes issues with `bash` on AIX and zOS.

Comment (keithc-ca) [reply ddr_artifacts.mk]: getmacros begins with
```sh
#!/bin/bash
```
and I think it does depend on features in `bash` that are not in `sh`.

Comment (charliegracie) [reply ddr_artifacts.mk]: makes sense then to use `bash`

**Pull request merged**